x/upgrade/keeper/keeper.go

```diff
@@ -28,16 +29,18 @@ type Keeper struct {
 	storeKey           sdk.StoreKey
 	cdc                codec.BinaryMarshaler
 	upgradeHandlers    map[string]types.UpgradeHandler
+	protoManager       baseapp.ProtocolManager
 }
 
 // NewKeeper constructs an upgrade Keeper
```

Comment:
Can we document all of the parameters here? I'd also like this to explicitly allow ProtocolManager to be nil.

Reply:
It doesn't seem like this has been addressed @technicallyty, although I guess documenting `vs ProtocolVersionSetter` would be sufficient.

Reply:
Does documentation on just the versionSetter suffice @aaronc? I can add comments to all of them, or maybe can do that in a separate PR?

Reply:
I would like comments on all of them. It can be a separate PR if you prefer.

Reply:
[added comments](https://github.com/cosmos/cosmos-sdk/pull/8897/commits/1df5a51989b4cbc17443418ea464f5adc3505f2e) for all the fields @aaronc

Reply:
That's great, but we also want docs for all arguments to `NewKeeper`. That is what users of the SDK will see in godocs when they're using `NewKeeper`.

Reply:
[added doc comments](https://github.com/cosmos/cosmos-sdk/pull/8897/files#diff-33ff09992f1a3583e2aa8eb932c62c833d52e446488070ed958eb672eed8da3eR35-R41). let me know if the format/content needs to change @aaronc